extensions/cornerstone-dynamic-volume/src/panels/PanelGenerateImage.tsx

```diff
@@ -0,0 +1,376 @@
+        for (const viewport of viewports) {
+          const viewportForRendering = cornerstoneViewportService.getCornerstoneViewport(
+            viewport.viewportId
+          );
+          viewportForRendering.render();
+        }
+      } else {
+        renderGeneratedImage(computedDisplaySet);
+      }
+    }
+  }
+
+  // if computedDisplaySet is defined, render when the data changes
+  useEffect(() => {
+    if (computedDisplaySet) {
+      renderGeneratedImage(computedDisplaySet);
+    }
+  }, [computedDisplaySet]);
+
+  function returnTo4D() {
+    const updatedViewports = hangingProtocolService.getViewportsRequireUpdate(
+      0,
+      uuidDynamicVolume.current
+    );
+    viewportGridService.setDisplaySetsForViewports(updatedViewports);
+  }
+
+  const handlePlay = () => {
+    const viewportInfo = cornerstoneViewportService.getViewportInfo(
+      viewports[activeViewportIndex].viewportId
+    );
+    const { element } = viewportInfo;
+    const cine = cines[activeViewportIndex];
+    if (!cine.isPlaying) {
+      cineService.playClip(element, { framesPerSecond: frameRate });
+      cine.isPlaying = true;
+    }
+  };
+
+  const handleStop = () => {
+    const viewportInfo = cornerstoneViewportService.getViewportInfo(
+      viewports[activeViewportIndex].viewportId
+    );
+    const { element } = viewportInfo;
+    const cine = cines[activeViewportIndex];
+    if (cine.isPlaying) {
+      cineService.stopClip(element);
+      cine.isPlaying = false;
+    }
+  };
+
+  const handleSetFrameRate = newFrameRate => {
+    if (newFrameRate >= 1 && newFrameRate <= 50) {
+      setFrameRate(newFrameRate);
+    }
+    const cine = cines[0];
+    if (cine.isPlaying) {
+      handleStop();
+      handlePlay();
+    }
+  };
+
+  function handleSliderChange(newValues) {
+    // setSliderValues(newValues);
+    const timeFrameValuesArray = Array.from(
+      { length: newValues[1] - newValues[0] + 1 },
+      (_, i) => i + newValues[0] - 1
+    );
+    setTimeFramesToUse(timeFrameValuesArray);
+  }
+
+  return (
+    <div className="flex flex-col">
+      <div className="flex flex-col p-4 space-y-4 bg-primary-dark">
+        <div className="marginBottom-10px">Frame Panel</div>
+        <div className="w-3">
+          <InputDoubleRange
+            labelClassName="text-black"
+            maxValue={rangeValues[1] || 2}
+            minValue={rangeValues[0] || 1}
+            onSliderChange={handleSliderChange}
+            step={10}
+            unit="%"
+            valueLeft={rangeValues[0] || 1}
+            valueRight={rangeValues[1] || 2}
+          />
+        </div>
+        <Select
+          label={t('Strategy')}
+          closeMenuOnSelect={true}
+          className="mr-2 bg-black border-primary-main text-white "
+          options={operationsUI}
+          placeholder={
+            operationsUI.find(option => option.value === options.Operation)
+              .placeHolder
+          }
```

Comment:
Do you really need to set the `placeholder`? I'm asking you because `value` prop should make it show the expected label, right?

Reply:
placeholder is what is shown in the select menu for each value currently